signalling/src/protocol/mod.rs

```diff
@@ -0,0 +1,126 @@
+/// The default protocol used by the signalling server
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(tag = "peer-type")]
+#[serde(rename_all = "lowercase")]
+/// Confirms registration
+pub enum RegisteredMessage {
```

Comment:
You're duplicating these definitions now or not? Best to put it into a single place :)